mlir/lib/Dialect/LLVMIR/IR/LLVMTypes.cpp

```diff
@@ -897,8 +821,7 @@ bool mlir::LLVM::isCompatibleVectorType(Type type) {
 
 Type mlir::LLVM::getVectorElementType(Type type) {
   return llvm::TypeSwitch<Type, Type>(type)
-      .Case<LLVMFixedVectorType, LLVMScalableVectorType, VectorType>(
-          [](auto ty) { return ty.getElementType(); })
+      .Case<VectorType>([](auto ty) { return ty.getElementType(); })
```

Comment:
```suggestion
      .Case([](VectorType ty) { return ty.getElementType(); })
```
Nit: Or maybe remove the switch altogether?